src/main/java/seedu/address/model/person/PersonWithinDateRangePredicate.java

```diff
@@ -0,0 +1,41 @@
+package seedu.address.model.person;
+
+import java.util.function.Predicate;
+
+public class PersonWithinDateRangePredicate implements Predicate<Person> {
+    private final int numberOfDays;
+
+    /**
+     * Constructs a {@code PersonWithinDateRangePredicate} object.
+     *
+     * @param days the number of days used as a range for this predicate.
+     */
+    public PersonWithinDateRangePredicate(int days) {
+        assert days >= 0 : "days should not be less than 0";
+        numberOfDays = days;
+    }
+    /**
+     * Evaluates this predicate on the given argument.
+     *
+     * @param person the input argument
+     * @return {@code true} if the input argument matches the predicate,
+     * otherwise {@code false}
+     */
+    @Override
+    public boolean test(Person person) {
```

Comment:
Maybe change the name of this method.

Reply:
This is an interface method so I can't change the method signature